core/geometry/src/polyface/PolyfaceQuery.ts

```diff
@@ -1349,6 +1434,25 @@ export class PolyfaceQuery {
     BuildAverageNormalsContext.buildFastAverageNormals(polyface, toleranceAngle);
   }
 
+  /**
+   * Offset the faces of the mesh.
+   * @param source original mesh
+   * @param signedOffsetDistance distance to offset
+   * @param offsetOptions angle options.  The default options are recommended.
+   * @returns shifted mesh.
+   */
+  public static cloneOffset(source: IndexedPolyface,
+    signedOffsetDistance: number,
+    offsetOptions: OffsetMeshOptions = OffsetMeshOptions.create()): Polyface {
```

Comment:
Why does this return `Polyface` instead of `IndexedPolyface`?

Reply:
I think the conversion to HalfEdgeGraph depends on being indexed, so the input side is probably correctly restricted.

On output, yes I'll switch to Indexed because that's what the builder returns.